src/main/java/dk/cngroup/lentils/service/StatusService.java

```diff
@@ -1,43 +1,65 @@
 package dk.cngroup.lentils.service;
 
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.Hint;
 import dk.cngroup.lentils.entity.Status;
 import dk.cngroup.lentils.entity.Team;
+import dk.cngroup.lentils.repository.CypherRepository;
+import dk.cngroup.lentils.repository.HintTakenRepository;
 import dk.cngroup.lentils.repository.StatusRepository;
+import dk.cngroup.lentils.repository.TeamRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 public class StatusService {
 
     private StatusRepository statusRepository;
+    private CypherRepository cypherRepository;
+    private TeamRepository teamRepository;
+    private HintTakenRepository hintTakenRepository;
 
     @Autowired
-    public StatusService(StatusRepository statusRepository) {
+    public StatusService(StatusRepository statusRepository, CypherRepository cypherRepository, TeamRepository teamRepository, HintTakenRepository hintTakenRepository) {
         this.statusRepository = statusRepository;
+        this.cypherRepository = cypherRepository;
+        this.teamRepository = teamRepository;
+        this.hintTakenRepository = hintTakenRepository;
     }
 
-    public void markCypherSolvedForTeam(Long cypherId, Long teamId) {
-        Status status = statusRepository.findByTeamIdAndCypherId(teamId, cypherId);
+    public void markCypherSolvedForTeam(Cypher cypher, Team team) {
+        Status status = statusRepository.findByTeamAndCypher(team, cypher);
         saveNewStatus(status, CypherStatus.SOLVED);
     }
 
-    public void markCypherSkippedForTeam(Long cypherId, Long teamId) {
-        Status status = statusRepository.findByTeamIdAndCypherId(teamId, cypherId);
+    public void markCypherSolvedForTeam(Long cypherId, Long teamId) {
+        Optional<Cypher> cypher = cypherRepository.findById(cypherId);
```

Comment:
Tady uz by si uz mela pouzivat primo cypherService a ne cypher repository. To cypherService uz by ti nemelo vracet Optional, ale primo ten objekt Cypher.